Proxy panic: duplicate Prometheus metrics registration when S3 and Swift are both configured

## Description

The proxy panics on startup with `duplicate metrics collector registration attempted` when both S3 and Swift storage types are configured.

## Stack trace

```
panic: duplicate metrics collector registration attempted

goroutine 1 [running]:
github.com/prometheus/client_golang/prometheus.(*Registry).MustRegister(...)
    prometheus/registry.go:406
github.com/prometheus/client_golang/prometheus/promauto.Factory.NewCounterVec(...)
    prometheus/promauto/auto.go:276
github.com/prometheus/client_golang/prometheus/promauto.NewCounterVec(...)
    prometheus/promauto/auto.go:176
github.com/clyso/chorus/service/proxy.Start.ProxyMiddleware.func9(...)
    pkg/metrics/http_middleware.go:45
github.com/clyso/chorus/service/proxy/router.New(...)
    service/proxy/router/server.go:88
```

## Root cause

`ProxyMiddleware()` in `pkg/metrics/http_middleware.go` returns a closure that registers Prometheus metrics (`promauto.NewCounterVec`, `promauto.NewHistogramVec`) each time it is invoked. When `router.New()` in `service/proxy/router/server.go` applies the metrics middleware once per storage type (S3 and Swift), the second invocation attempts to re-register the same metric names (`proxy_requests_total`, `proxy_response_status`, `proxy_response_time_seconds`), causing the panic.

## Affected versions

v0.7.4 (and likely earlier versions when both S3 and Swift storages are configured simultaneously)

## Fix

Move the `promauto.New*` calls from inside the returned closure to the body of `ProxyMiddleware()`, so metrics are registered once regardless of how many storage types use the middleware.

--- a/pkg/metrics/http_middleware.go
+++ b/pkg/metrics/http_middleware.go
@@ -40,60 +40,60 @@ func (rw *responseWriter) WriteHeader(code int) {
 	rw.ResponseWriter.WriteHeader(code)
 }
 
+var proxyTotalRequests = promauto.NewCounterVec(
+	prometheus.CounterOpts{
+		Name: "proxy_requests_total",
+		Help: "Number of requests to chorus s3 proxy.",
+	},
+	[]string{"method"},
+)
+
+var proxyResponseStatus = promauto.NewCounterVec(
+	prometheus.CounterOpts{
+		Name: "proxy_response_status",
+		Help: "Status of chorus s3 proxy response.",
+	},
+	[]string{"status"},
+)
+
+var proxyHttpDuration = promauto.NewHistogramVec(prometheus.HistogramOpts{
+	Name:    "proxy_response_time_seconds",
+	Help:    "Duration of chorus s3 proxy requests.",
+	Buckets: prometheus.DefBuckets,
+}, []string{"method"})
+
 func ProxyMiddleware() func(next http.Handler) http.Handler {
 	return func(next http.Handler) http.Handler {
-		var totalRequests = promauto.NewCounterVec(
-			prometheus.CounterOpts{
-				Name: "proxy_requests_total",
-				Help: "Number of requests to chorus s3 proxy.",
-			},
-			[]string{"method"},
-		)
-
-		var responseStatus = promauto.NewCounterVec(
-			prometheus.CounterOpts{
-				Name: "proxy_response_status",
-				Help: "Status of chorus s3 proxy response.",
-			},
-			[]string{"status"},
-		)
-
-		var httpDuration = promauto.NewHistogramVec(prometheus.HistogramOpts{
-			Name:    "proxy_response_time_seconds",
-			Help:    "Duration of chorus s3 proxy requests.",
-			Buckets: prometheus.DefBuckets,
-		}, []string{"method"})
-
 		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 			method := xctx.GetMethod(r.Context())
 
-			timer := prometheus.NewTimer(httpDuration.WithLabelValues(method.String()))
+			timer := prometheus.NewTimer(proxyHttpDuration.WithLabelValues(method.String()))
 			rw := NewResponseWriter(w)
 			next.ServeHTTP(rw, r)
 			statusCode := rw.statusCode
 
-			responseStatus.WithLabelValues(strconv.Itoa(statusCode)).Inc()
-			totalRequests.WithLabelValues(method.String()).Inc()
+			proxyResponseStatus.WithLabelValues(strconv.Itoa(statusCode)).Inc()
+			proxyTotalRequests.WithLabelValues(method.String()).Inc()
 			timer.ObserveDuration()
 		})
 	}
 }
 
-func AgentMiddleware(next http.Handler) http.Handler {
-	var totalRequests = promauto.NewCounterVec(
-		prometheus.CounterOpts{
-			Name: "agent_requests_total",
-			Help: "Number of requests to chorus agent.",
-		},
-		[]string{"status"},
-	)
+var agentTotalRequests = promauto.NewCounterVec(
+	prometheus.CounterOpts{
+		Name: "agent_requests_total",
+		Help: "Number of requests to chorus agent.",
+	},
+	[]string{"status"},
+)
 
+func AgentMiddleware(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 
 		rw := NewResponseWriter(w)
 		next.ServeHTTP(rw, r)
 		statusCode := rw.statusCode
 
-		totalRequests.WithLabelValues(strconv.Itoa(statusCode)).Inc()
+		agentTotalRequests.WithLabelValues(strconv.Itoa(statusCode)).Inc()
 	})
 }

